src/common.js

```diff
@@ -1,16 +1,62 @@
+import path from 'path';
+import url from 'url';
+
+import fileUrlFromPath from 'file-url';
+import SourceMapSupport from 'source-map-support';
+
 export const hasSelf = typeof self !== 'undefined';
 
 const envGlobal = hasSelf ? self : global;
 export { envGlobal as global };
 
-export let baseUrl;
-if (typeof location !== 'undefined') {
-  baseUrl = location.href.split('#')[0].split('?')[0];
-  const lastSepIndex = baseUrl.lastIndexOf('/');
-  if (lastSepIndex !== -1)
-    baseUrl = baseUrl.slice(0, lastSepIndex + 1);
+
+function getDefaultBaseUrl() {
+  let url;
+
+  if (typeof location !== 'undefined') {
+    url = location.href.split('#')[0].split('?')[0];
+    const lastSepIndex = url.lastIndexOf('/');
+    if (lastSepIndex !== -1) {
+      url = url.slice(0, lastSepIndex + 1);
+    }
+  }
+
+  return url;
+}
+
+export let baseUrl = getDefaultBaseUrl();
+
+
+export function urlToPath(url) {
+  return new URL(url).pathname;
 }
 
+export function basename(moduleUrl) {
+  const filePath = urlToPath(moduleUrl);
+  return path.basename(filePath);
+}
+
+export function dirname(moduleUrl) {
+  const filePath = urlToPath(moduleUrl);
+  return path.dirname(filePath);
+}
+
+export const sourceMapSources = {};
+
+SourceMapSupport.install({
+  retrieveSourceMap: function(source) {
+    if (!sourceMapSources[source])
+      return null;
+
+    return {
+      url: source.replace('!transpiled', ''),
+      map: sourceMapSources[source]
+    };
+  }
+});
```

Comment:
We don't want this to apply to the browser build surely? Rather this stuff seems like it belongs in a `src/system-node.js` or similar.

Reply:
Since babel is being used to transpile cjs->system, sourcemaps are needed, and this helps chrome make proper use of them. Since both Node and Chrome both make use of the same V8 runtime, this ends up being done for both.